docs/decisions/0001-namespacing-branches.rst

```diff
@@ -0,0 +1,84 @@
+Namespacing Branches and Tags
+#############################
+
+Status
+******
+
+Draft
+
+Context
+*******
+
+This ADR was written in response to this `Discourse thread`_.
+
+Currently, 2U continuous delivery tooling creates branches and tags
+in the repository "global namespace."  By that I mean that branches
+and tags have the form ``release`` or ``release-candidate-3729``.
+This is confusing because those branches and tags appear to be global,
+officially sanctioned by the project, and authoritative.
+
+A result of this is that the official named releases of the platform
+require a namespace specifier, for example,
+``open-release/olive.master``.
+
+This is confusing to most members of the community and reflects the
+prior stewardship arrangements for the project.
+
+It is not typical of open-source projects to allow participating
+organizations to push organization specific branches and tags to the
+canonical upstream.  However, we recommend not prohibiting this at this
+time.  There are two reasons for that:
+
+1. This would require more substantial changes to participant deployment
+   systems that would delay getting value from this initial change.
+2. The 2U branches and tags represent code that is battle-tested in
+   production at scale and are, thus, valuable sign posts for
+   community members.
+
+.. _Discourse thread: https://discuss.openedx.org/t/should-we-rename-the-release-branches/8827/7
+
+
+Decisions
+*********
+
+1. We will not forbid project participants with write access to
+   repositories from pushing branches or tags to the canonical
+   upstream hosted at the openedx GitHub organization
+2. We will require that project participants namespace their branches
+   and tags to clarify their origin and purpose
+3. The global namespace will be reserved for global project purposes
+
+Going forward branches and tags that originate from any specific
+organization partipatinging in the Open edX project must by namespaced
+as follows:
+
+``acme/release``
+
+or
+
+``acme/release-candidate-3729``
+
+In the case of the global namespace, tags and branches may still be
+further qualified to provide additional information and specificity.
+
+For example:
+
+``release`` would be consider in the global namespace.
+
+``release/palm`` would represent the branch for an officially
+sanctioned Open edX named release.
+
+
+Consequences
+************
+
+1. Continuous delivery systems and other automation should be updated
+   across the ecosystem to use the proper naming convention for
+   branches and tags
+2. Existing branches and tags should be updated to use the new naming
+   protocol
+3. There will be no change to the guidance that personal branches
+   should be namespaced by GitHub username.  For example, the pull
+   request for this ADR originated in the branch ``e0d/namespace-adr``
+4. A reasonable grace period should be established to schedule and
+   complete the required work
```

Comment:
@e0d @feanil @robrap 
```suggestion
4. The community will be informed that all personal and organizational branches
   and tags must be namespaced before 2024-03-11. After that point, Open edX
   project administrators may begin deleting un-namespaced branches and tags,
   particularly those which interfere with community processes such as the named
   release.
```

Reply:
Thanks for the update. 
FYI: @jmbowman: In case you want to discuss this deadline.

Reply:
This seems good to me.

Reply:
I'm going to leave this suggestion open for comments, but I plan to apply it before merging.

Reply:
@kdmccormick: Are you planning on scripting a large clean-up, or just proposing clean-up of specific known problematic branches or tags? I'm unclear how much work will be involved, if any, and that would dictate how problematic any deadline might be. Do you have information around this? Thank you.

Reply:
What we will clean up in March are branches and tags matching `release` or `release/*`.

We may want to clean up other branches and tags in the future, but we wouldn't do so without a separate heads-up and grace period.

Reply:
This last comment of yours is much more targeted and less scary than the text of the ADR, especially since you've produced a list of potential target branches. Can you update the ADR to match?

Reply:
As mentioned in https://github.com/openedx/.github/pull/36#discussion_r1445157088, I've gone the other direction and removed details. Announcements to follow with clearly state which branches up for deletion and what the grace period is.